src/coordinator/executor/transform/cache.go

```diff
@@ -70,3 +71,92 @@ func (c *BlockCache) Get(key parser.NodeID) (block.Block, bool) {
 	b, ok := c.blocks[key]
 	return b, ok
 }
+
+type cacheTime int64
+
+func fromTime(t time.Time) cacheTime {
+	return cacheTime(t.UnixNano())
+}
+
+func (c cacheTime) toTime() time.Time {
+	return time.Unix(0, int64(c))
+}
+
+// TimeCache is used to cache blocks over time. It also keeps track of blocks which have been processed
+type TimeCache struct {
+	blocks          map[cacheTime]block.Block
+	processedBlocks map[cacheTime]bool
```

Comment:
make this a map of struct{} rather than bool

Reply:
Alternatively, make a struct which contains a block.Block as well as a flag for `processed`, and combine these 2 maps?

Reply:
no longer valid since i switched to list